src/gui/editors/PianoRoll.cpp

```diff
@@ -863,6 +863,10 @@ void PianoRoll::setCurrentMidiClip( MidiClip* newMidiClip )
 	{
 		Engine::getSong()->playMidiClip( nullptr );
 	}
+	else if (Engine::getSong()->isPaused())
```

Comment:
I wonder if it's sufficient to do this in `Engine::getSong()->isPaused() && Engine::getSong()->playMode() == Song::Mode_PlayMidiClip`  cases. If it is, this can be merged with the `if` statement above.

Reply:
This has a undesirable side-effect: Opening a MIDI clip in piano roll when the song is paused will change the playback state from paused to stopped, by calling `stop` in `playMidiClip`.
I think the correct solution is removing `Engine::getSong()->isPlaying()` from the `if` state above.

Reply:
To reproduce:
1. Add a MIDI clip
2. Set the playhead mode to `|<<` or `<<`
3. Play the song in the song editor
4. Pause
5. Open the MIDI clip
6. The playhead will jump back as if the stop button is pressed